1-Draft/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,141 @@
+---
+RFC: 
+Author: Aditya Patwardhan
+Status: Draft
+SupercededBy: 
+Version: 0.1
+Area: Formatting and Output
+Comments Due: 03/26/2018
+Plan to implement: Yes
+---
+
+# Native Markdown Rendering
+
+Render markdown content on console to improve readability. 
+
+## Motivation
+
+Markdown is a common authoring format used by the community.
+There is no easy way in PowerShell to visualize a markdown document on console.
+
+## Specification
+
+This RFC proposes to use VT100 escape sequences to render markdown content.
+`ConvertFrom-Markdown` cmdlet as part of `Microsoft.PowerShell.MarkdownRender` PowerShell module would consume a string or a file path and output a VT100 encoded string.
+Optionally, the output can be formatted as HTML.
+For converting strings to VT100 coded strings, we will be writing an extension to [markdig](https://github.com/lunet-io/markdig).
+The extension will insert VT100 escape sequences as appropriate.
+
+```PowerShell
+ConvertFrom-Markdown [-Path] <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+
+ConvertFrom-Markdown -LiteralPath <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+
+ConvertFrom-Markdown -InputObject <psobject> -AsHTML -AsPlainText  [<CommonParameters>]
+```
+
+### Parameters
+
+- **Path** : Accepts an array of file paths with markdown content.
+- **LiteralPath** : Accepts an array of literal paths with markdown content.
+- **InputObject** : Accepts an InputObject of `System.IO.FileSystemInfo`, `string` type.
+- **AsHTML** : When selected, the output will be HTML.
+- **AsPlainText** : When selected, the output will be plain text.
+
+The default output will be a string encoded with VT100 escape sequences.
+
+## Supported Markdown Elements
+
+We will be supporting a limited set of markdown elements in the initial version.
```

Comment:
For the completeness of the RFC I would expect that we will describe the escape sequences for all markdown elements to have an accurate parity with that the markdig supports for HTML. (We are free implement its step-by-step).

Reply:
The RFC describes escape sequences for all the markdown elements that will be supported in this version. Further work on the extension might have another RFC, if the changes are significant enough.

Reply:
I agree if changing already implemented escapes will not be a breaking change.